src/wasm/wasm-validator.cpp

```diff
@@ -431,8 +431,17 @@ void WasmValidator::visitUnary(Unary *curr) {
       shouldBeTrue(curr->value->type == i64, curr, "i64.eqz input must be i64");
       break;
     }
-    case ExtendSInt32:           shouldBeEqual(curr->value->type, i32, curr, "extend type must be correct"); break;
-    case ExtendUInt32:           shouldBeEqual(curr->value->type, i32, curr, "extend type must be correct"); break;
+    case ExtendSInt32:
```

Comment:
Perhaps the original extend instructions should be renamed now?

Reply:
Maybe. Actually a lot of instructions probably should be renamed, and probably along with the binary format opcodes, it's kind of a mess (a lot of the opcode names predate the current spec scheme). I almost renamed the extension instructions here but decided that's not a great idea to mix it in with a functionality PR.

Reply:
Makes sense, probably better for a cleanup PR.

Reply:
Agreed, let's do a bulk renaming eventually to the spec names, but not here.